src/Sparrow/Json/BlittableJsonTextWriter.cs

```diff
@@ -170,16 +169,63 @@ public unsafe void WriteString(LazyStringValue str)
             if (numberOfEscapeSequences == 0)
             {
                 WriteRawString(strBuffer, size);
+            }
```

Comment:
No, this was written intentionally in this way, so there would be no jumps in the code in the common case.